runtime/Stdlib_Array.res

```diff
@@ -103,11 +103,14 @@ external copyWithin: (array<'a>, ~target: int, ~start: int, ~end: int) => array<
 @send external shift: array<'a> => option<'a> = "shift"
 
 @variadic @send
-external splice: (array<'a>, ~start: int, ~remove: int, ~insert: array<'a>) => unit = "splice"
+external splice: (array<'a>, ~start: int, ~remove: int, ~insert: array<'a>) => array<'a> = "splice"
```

Comment:
I think we willingly ignore the results of functions that mutate their arguments to make it clear that they have side effects, I'd likely rename it to something more explicit otherwise
```suggestion
external spliceInPlace: (array<'a>, ~start: int, ~remove: int, ~insert: array<'a>) => array<'a> = "splice"
```

Reply:
Yes, non breaking and the current bindings do have a purpose. So I agree with this comment.

Reply:
Do you mean to keep the current `splice` as it is and add a new bindings with the return type?

Reply:
yeah I'd add a new external with the return type.

Reply:
Is it OK to have the version in rescript breaks some backward compatibility? If so I think making Array.splice behave as JS does would provide a smoother experience. Unlike Array.sort the current binding of Array.splice is more restricting and less powerful. For Array.sort one can often fall back on toSorted but splice is different.

Reply:
I removed the change from the PR. It's ready for review now.